scripts/bootstrap/buildbootstrapcli.sh

```diff
@@ -0,0 +1,271 @@
+    local maxversionmid=0
+    local maxversionlo=0
+    local maxversiontag
+    local versionrest
+    local versionhi
+    local versionmid
+    local versionlo
+    local versiontag
+    local foundmax
+
+    for d in $1/*; do
+
+        if [[ -d $d ]]; then
+            versionrest=$(basename $d)
+            versionhi=${versionrest%%.*}
+            versionrest=${versionrest#*.}
+            versionmid=${versionrest%%.*}
+            versionrest=${versionrest#*.}
+            versionlo=${versionrest%%-*}
+            versiontag=${versionrest#*-}
+            if [[ $versiontag == $versionrest ]]; then
+                versiontag=""
+            fi
+
+            foundmax=0
+
+            if [[ $versionhi -gt $maxversionhi ]]; then
+                foundmax=1
+            elif [[ $versionhi -eq $maxversionhi ]]; then
+                if [[ $versionmid -gt $maxversionmid ]]; then
+                    foundmax=1
+                elif [[ $versionmid -eq $maxversionmid ]]; then
+                    if [[ $versionlo -gt $maxversionlo ]]; then
+                    foundmax=1
+                    elif [[ $versionlo -eq $maxversionlo ]]; then
+                        # tags are used to mark pre-release versions, so a version without a tag
+                        # is newer than a version with one.
+                        if [[ "$versiontag" == "" || $versiontag > $maxversiontag ]]; then
+                            foundmax=1
+                        fi
+                    fi
+                fi
+            fi
+
+            if [[ $foundmax != 0 ]]; then
+                maxversionhi=$versionhi
+                maxversionmid=$versionmid
+                maxversionlo=$versionlo
+                maxversiontag=$versiontag
+            fi
+        fi
+    done
+
+    echo $maxversionhi.$maxversionmid.$maxversionlo${maxversiontag:+-$maxversiontag}
+}
+
+__build_arch=x64
+__build_os=Linux
+__runtime_id=
+__corelib=
+__configuration=debug
+__clangversion=
+
+while [[ "$1" != "" ]]; do
+    lowerI="$(echo $1 | awk '{print tolower($0)}')"
+    case $lowerI in
+    -h|--help)
+        usage
+        exit 1
+        ;;
+    -arch)
+        shift
+        __build_arch=$1
+        ;;
+    -rid)
+        shift
+        __runtime_id=$1
+        ;;
+    -os)
+        shift
+        __build_os=$1
+        ;;
+    -debug)
+        __configuration=debug
+        ;;
+    -release)
+        __configuration=release
+        ;;
+    -corelib)
+        shift
+        __corelib=$1
+        ;;
+    -seedcli)
+        shift
+        __seedclipath=$1
+        ;;
+    -clang)
+        shift
+        __clangversion=clang$1
+        ;;
+     *)
+    echo "Unknown argument to build.sh $1"; exit 1
+    esac
+    shift
+done
+
+# NOTE: when realpath is not present, use readlink -e or readlink --canonicalize depending on the platform
+__seedclipath=`realpath $__seedclipath`
+
+mkdir -p $__runtime_id-$__build_arch
+cd $__runtime_id-$__build_arch
+
+if [[ -d dotnetcli ]]; then
```

Comment:
this path is used throughout, consider making it a variable and/or making the path absolute

Reply:
Which path do you mean? It is not clear from the comment location.

Reply:
the relative "dotnetcli" directory could also be defined as a variable to an absolute path which may make changes down the road slightly easier to deal with.  Not a big deal